becquerel/core/spectrum.py

```diff
@@ -438,16 +451,22 @@ def __add__(self, other):
         Raises:
           TypeError: if other is not a Spectrum
           SpectrumError: if spectra are different lengths,
-            or if only one is calibrated
+            if only one is calibrated or if specra have not both
+            been initialized with counts or CPS, respectively.
           NotImplementedError: if spectra are calibrated differently
 
         Returns:
           a summed Spectrum object
         """
 
         self._add_sub_error_checking(other)
+        if (self._counts is None) ^ (other._counts is None):
+            raise SpectrumError(
+                'Spectrum addition with a mix of counts and CPS is ' +
+                'ambiguous, use Spectrum(counts=specA.counts+specB.counts)' +
+                ' or Spectrum(cps=specA.cps+specB.cps) instead.')
 
-        if self.counts is not None and other.counts is not None:
+        if self._counts is not None and other._counts is not None:
```

Comment:
PyLint complains: 'Access to a protected member _counts of a client class (protected-access)'. If that is an issue we need a method to check for counts vs cps based spectra.